methods/SPIA/SPIA.py

```diff
@@ -0,0 +1,292 @@
+    across the pathway topology.These two aspects are captured by two independent probability values (pNDE and pPERT)
+    and are are finally combined into one global probability value: pG.
+
+    Pipeline schema:
+        1.  Gene expression change between two conditions (fold change) is calculated
+        2.  Differentially expressed genes are identified.
+        3.  KEGG Pathways database is searched for pathways containing at least one differentially expressed gene.
+        4.  pNDE is calculated for each pathway
+        5.  Acc(net perturbation accumulation) is calculated for each pathway
+        6.  pPERT is calculated based on the amount of perturbation measured in each pathway
+        7.  pG is calculated for each pathway
+        8.  Multiple hypothesis testing adjustments of each pathway significance are performed by FDR
+        and Bonferroni correction calculation
+
+    Additional method arguments allow specifying organism for database selection ,
+    threshold for identification of differentially expressed genes,
+    number of iterations of random sampling part of algorithm
+    or comma separated list of values for gene relations (if other than default is needed)
+
+    For more information, please refer to:
+    Adi Laurentiu Tarca, Sorin Draghici, Purvesh Khatri, Sonia S. Hassan, Pooja Mittal,
+    Jung-sun Kim, Chong Jai Kim, Juan Pedro Kusanovic, Roberto Romero;
+    A novel signaling pathway impact analysis, Bioinformatics,
+    Volume 25, Issue 1, 1 January 2009, Pages 75–82
+
+    """
+    help = __doc__
+    name = "SPIA"
+
+    def __init__(self, organism: str = 'hsa', threshold: float = 0.05, nB: int = 2000, beta=None, markdown: str = ''):
+        """
+
+        Args:
+            organism: organism name (ex. 'Homo sapiens', 'human', 'hsa')
+            threshold: float: threshold for identification of differentially expressed genes
+            nB: number of iterations of random sample choosing at SPIA algorithm
+            beta: list of gene relations values, if None the values are default
+            markdown: generate additional markdown output file with given name
+        """
+        for x in SPECIES:
+            if organism in x:
+                self.organism = x[1]
+                break
+        else:
+            raise Exception("Unknown organism")
+        if threshold < 0 or threshold > 1:
+            raise ValueError('Indices need to be in (0,1) range')
+        self.threshold = threshold
+        self.nB = nB
+        self.beta = beta
+        self.markdown = markdown
+        if markdown:
+            if os.path.exists(markdown if '.md' in markdown else markdown.split('.')[0] + '.md'):
+                print("Warning: '" + markdown + "' file already exists and will be overwritten!")
+
+    @staticmethod
+    def load_data_dict(dictionary, all):
+        '''
+
+        Part of this code is imported from https://github.com/iseekwonderful/PyPathway on MIT license.
```

Comment:
PyPathway is **not** MIT licensed. It appears that it is (and always was) GNU GPL 3.0. This requires further consideration, as GNU GPL derived cannot be used in (our) MIT project. However, we could change our license to GPL 3.0

Moreover, I think that this one requires more visible attribution:
1. on top of the file:
```
Parts of this implementation incorporate, build upon or adapt PyPathway (github.com/iseekwonderful/PyPathway) code by iseekwonderful
[insert correct licence information here + link to the full text of the licence]
``` 
2. in the method docstring